File: solutions/prospective/algorithms/CLRS/c/linear_search.c
/*
Linear Search

Time Complexity: O(n)
Space Complexity: O(1)

Returns the index of target in arr, or -1 if not found.
*/
// === MEMO START ===
int linear_search(int *arr, int n, int target) {
    for (int i = 0; i < n; i++) {
        if (arr[i] == target) {
            return i;
        }
    }
    return -1;
}

File: solutions/prospective/algorithms/CLRS/js/linear_search.js
/**
 * Linear Search
 * 
 * Time Complexity: O(n)
 * Space Complexity: O(1)
 * 
 * Returns the index of target in arr, or -1 if not found.
 */
// === MEMO START ===
function linearSearch(arr, target) {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === target) {
            return i;
        }
    }
    return -1;
}

